### DISCUSSION TITLE
Add operator support to Timestamptz updates

### DISCUSSION THREAD

**Discussion by theelderbeever** [piccolo-orm/piccolo#483] [Ideas]
Similar to how it is shown in the [docs](https://piccolo-orm.readthedocs.io/en/latest/piccolo/query_types/update.html#integer-columns) for numeric types it would be awesome to be able to perform update operations using `Timestamptz` columns.

```python
class Tasks(Table):
  started_at = Timestamptz()
  completed_at = Timestamptz(null=True)
  runtime = Numeric(null=True)

await Tasks.insert(Tasks(started_at=datetime.now()))

completed_at = datetime.now()
await Tasks.update(
  {
    "completed_at": completed_at, 
    "runtime": (completed_at - Tasks.started_at).total_seconds()}
)
```

So the basic idea being that updates treat values as `datetime` objects possibly? Maybe there is a separate way around this.

**Comment by dantownsend** [comment]
It's a good idea. The easiest to implement initially would be adding / substracting `timedelta`:

```python
await Task.update({Task.completed_at: Task.completed_at + datetime.timedelta(days=1)})
```

**Reply by dantownsend** [reply]
I've made a prototype for this - https://github.com/piccolo-orm/piccolo/pull/486

**Comment by dantownsend** [comment]
I've implemented a first version of this - will release it now.
